stdlib/map.mli

```diff
@@ -283,7 +283,24 @@ module type S =
     (** Same as {!Map.S.map}, but the function receives as arguments both the
        key and the associated value for each binding of the map. *)
 
+    (** {6 Iterators} *)
 
+    val to_seq : 'a t -> (key * 'a) Seq.t
+    (** Iterate on the whole map, in ascending order
+        @since NEXT_RELEASE *)
+
+    val to_seq_at : key -> 'a t -> (key * 'a) Seq.t
```

Comment:
I agree `to_seq_from` is more descriptive than `to_seq_at`.  But we could also incorporate more general slicing in the main sequence extractor,
```ocaml
val to_seq : ?first: (key -> bool) -> ?last: (key -> bool) -> 'a t -> (key * 'a) Seq.t
```
borrowing the semantics of `find_first` and `find_last`. While the end-of-sequence can be adjusted with a sequence combinator, direct support for it has the advantage of only holding on to the part of the container it needs plus `O(log(n)`.